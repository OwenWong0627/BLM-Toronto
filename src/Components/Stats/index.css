.stats-container {
   display: flex;
   flex-direction: row;
   justify-content: space-evenly;
   background-color: rgba(0, 132, 244, 0.05);
   padding: 75px 10vw;
}
.stat-box {
   text-align: center;
   margin-top: 25px;
   margin-bottom: 25px;
   width: 300px;
}

.stat-number {
   font-size: 48px;
   line-height: 1.25;
   font-weight: 700;
   font-family: 'Trebuchet MS';
   color: rgb(0, 62, 107);
}

.stat-text {
   font-size: 24px;
   line-height: 1.25;
   font-weight: 600;
   font-family: 'Trebuchet MS';
   color: rgb(98, 125, 152);
}

@media screen and (max-width: 960px) {
   .stats-container {
      display: flex;
      flex-direction: column;
      align-items: center;
      padding: 50px 10vw;
   }
   .stat-number {
      font-size: 36px;
   }
   
   .stat-text {
      font-size: 20px;
   }
}
/* @media screen and (min-width: 48em){
   .stat-box {
    margin-top: 0px;
    margin-bottom: 0px;
    width: 200px;
   }
} */