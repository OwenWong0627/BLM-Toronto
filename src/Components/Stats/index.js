import React from 'react';
import './Stats.css';

/**
 * This component represents the Stats component under the Hero Component and above the Card Component
 * This component is completely static and only contains some text
 */
function Stats() {
   return (
      <div className="stats-container">
         <div className="stat-box" data-testid="stats-1">
            <h2 className="stat-number">100+</h2>
            <h2 className="stat-text">Businesses Included</h2>
         </div>
         <div className="stat-box" data-testid="stats-2">
            <h2 className="stat-number">10,000+</h2>
            <h2 className="stat-text">Searches Made</h2>
         </div>
      </div>
   )
}

export default Stats;
