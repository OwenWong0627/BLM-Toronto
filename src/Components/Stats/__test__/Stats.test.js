import React from 'react';
import ReactDOM from 'react-dom';
import Stats from '..';

import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';

afterEach(cleanup);
it('Renders without crashing', () => {
   const div = document.createElement("div");
   ReactDOM.render(<Stats />, div);
});

it('Renders the correct content', () => {
   const { getByTestId } = render(<Stats />);
   expect(getByTestId("stats")).toContainHTML("h2");
   expect(getByTestId("stats")).toHaveTextContent("100+");
   expect(getByTestId("stats")).toHaveTextContent("Businesses Included");
   expect(getByTestId("stats")).toHaveTextContent("10,000+");
   expect(getByTestId("stats")).toHaveTextContent("Searches Made");
});

it('matches snapshot', () => {
   const tree = renderer.create(<Stats />).toJSON();
   expect(tree).toMatchSnapshot();
});