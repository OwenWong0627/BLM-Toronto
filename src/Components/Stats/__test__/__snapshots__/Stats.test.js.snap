// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="stats-container"
  data-testid="stats"
>
  <div
    className="stat-box"
  >
    <h2
      className="stat-number"
    >
      100+
    </h2>
    <h2
      className="stat-text"
    >
      Businesses Included
    </h2>
  </div>
  <div
    className="stat-box"
  >
    <h2
      className="stat-number"
    >
      10,000+
    </h2>
    <h2
      className="stat-text"
    >
      Searches Made
    </h2>
  </div>
</div>
`;
